Automattic/themes
# Blockbase: Register Fonts via Webfonts API instead of theme.json

#### Changes proposed in this Pull Request:

This PR changes the way Blockbase registers fonts, from theme.json to use the Webfonts API directly instead.  

This is done so that we can conditionally load them — only if Jetpack fonts are **not** active.

**To test**:

- Checkout this PR, activate Blockbase locally (no other plugins active), ensure custom fonts load as expected in the editors and front-end:

<img width="1512" alt="Screenshot 2022-12-21 at 11 39 16 AM" src="https://user-images.githubusercontent.com/5375500/208991056-a76f5cfa-4bf3-4115-90e4-d68e98fc3a6b.png">

- Checkout this version of blockbase onto a wpcom simple site (you can use `npm run deploy:push:blockbase`, verify there are no duplicates and you see all the Jetpack fonts, e.g. with Zoologist:

<img width="1512" alt="Screenshot 2022-12-21 at 11 52 36 AM" src="https://user-images.githubusercontent.com/5375500/208991631-114874a4-5b00-4c13-9d1c-80ea3feeae08.png">

#### Related issue(s):

Solves #6765, alternative to #6777 which failed for reasons I still can't quite identify.

#### More reasoning why

Quoting @simison [here](https://github.com/Automattic/themes/pull/6777#pullrequestreview-1219020200) for why I went with this approach instead:

> Although I think we should still just disable Blockbase fonts entirely when Jetpack fonts module is active instead of doing this kind of "merge both sources" stuff, as Jetpack fonts:
> - works with all themes and experience is hence more consistent especially considering theme switches,
> - is a single source that's easier to keep up to date. Blockbase seems to already have outdated fonts list. Current either/or setup is way too complex, and this PR is actually making it even more complicated (merged lists) although it does fix an issue.
>- Jetpack fonts are now loaded from WP.com instead of Google
